RFS/src/main/java/org/opensearch/migrations/bulkload/common/OpenSearchClient.java

```diff
@@ -69,23 +79,43 @@ protected OpenSearchClient(RestClient client, FailedRequestsLogger failedRequest
     }
 
     public Version getClusterVersion() {
-        return client.getAsync("", null)
+        var versionFromRootApi = client.getAsync("", null)
             .flatMap(resp -> {
-                try {
-                    return Mono.just(versionFromResponse(resp));
-                } catch (Exception e) {
-                    return Mono.error(new OperationFailed(e.getMessage(), resp));
+                if (resp.statusCode == 200) {
+                    return versionFromResponse(resp);
+                }
+                // If the root API doesn't exist, the cluster is OpenSearch Serverless
+                if (resp.statusCode == 404) {
+                    return Mono.just(AMAZON_SERVERLESS_VERSION);
                 }
```

Comment:
This is a pretty non-intuitive conclusion to reach just from performing a GET on the default REST path.  How confident are we that this is a durable condition?

Reply:
Pretty confident.  The root API was intentionally removed from OpenSearch Serverless, so adding it back in would be very unexpected, it surfaces a lot of information that would be hard to rationalize - there is no option to get this information today.

I've created [MIGRATIONS-2209](https://opensearch.atlassian.net/browse/MIGRATIONS-2209) to track using an api and internally made a request for an API that indicates the wire compatibility version, when I've got the receipt for that request I'll add it to that task.